src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs

```diff
@@ -1,79 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
-using System.Diagnostics;
-using Microsoft.Azure.WebJobs.Host;
-using Microsoft.Azure.WebJobs.Logging;
-using Microsoft.Azure.WebJobs.Script.Config;
-
-namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
-{
-    public class SystemTraceWriter : TraceWriter
-    {
-        private IEventGenerator _eventGenerator;
-        private ScriptSettingsManager _settingsManager;
-        private string _appName;
-        private string _subscriptionId;
-
-        public SystemTraceWriter(TraceLevel level) : this(new EventGenerator(), ScriptSettingsManager.Instance, level)
-        {
-        }
-
-        public SystemTraceWriter(IEventGenerator eventGenerator, ScriptSettingsManager settingsManager, TraceLevel level) : base(level)
-        {
-            _settingsManager = settingsManager;
-            _appName = _settingsManager.AzureWebsiteUniqueSlotName;
-            _subscriptionId = Utility.GetSubscriptionId();
-            _eventGenerator = eventGenerator;
-        }
-
-        public override void Trace(TraceEvent traceEvent)
-        {
-            // Apply standard event properties
-            // Note: we must be sure to default any null values to empty string
-            // otherwise the ETW event will fail to be persisted (silently)
-            string subscriptionId = _subscriptionId ?? string.Empty;
-            string appName = _appName ?? string.Empty;
-            string source = traceEvent.Source ?? string.Empty;
-            string summary = Sanitizer.Sanitize(traceEvent.Message) ?? string.Empty;
-
-            // Apply any additional extended event info from the Properties bag
-            string functionName = string.Empty;
-            string eventName = string.Empty;
-            string details = string.Empty;
-            if (traceEvent.Properties != null)
-            {
-                object value;
-                if (traceEvent.Properties.TryGetValue(ScriptConstants.TracePropertyIsUserTraceKey, out value)
-                    && value is bool && (bool)value == true)
-                {
-                    // we don't write user traces to system logs
-                    return;
-                }
-
-                if (traceEvent.Properties.TryGetValue(ScriptConstants.TracePropertyFunctionNameKey, out value) && value != null)
-                {
-                    functionName = value.ToString();
-                }
-
-                if (traceEvent.Properties.TryGetValue(ScriptConstants.TracePropertyEventNameKey, out value) && value != null)
```

Comment:
Does anyone know where `TracePropertyEventNameKey` is ever set? I don't see it used anywhere.

Reply:
Looks like I added that when I added system tracing initially https://github.com/Azure/azure-webjobs-sdk-script/commit/cb7e7411d3beb7ff3d701640795a7cfa41ec3c53#diff-5d1f99d205acac1fcab3872af62b69c0

Event names get used when logging metric events I believe. See SystemMetricEvent

Reply:
In that commit, I only see this property added in tests. I see it read and applied, but I couldn't see anywhere that we wrote to it. Did we use it in the product anywhere? It looks like SystemMetricEvent carries this value in its own `EventName` property.

If you do a 'Find all references' in VS, it's only written to in tests. If we don't use it anywhere, I can remove it in this PR.

Reply:
Unless it is causing you problems, I'd say just leave it and I'll deal with it separately